apps/dashboard/lib/get-all-transactions.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  fetchFirstTransactions,
+  fetchPaginatedTransactions,
+} from "@/services/graphql/queries/get-transactions"
+
+export async function fetchAllTransactionsTillDate({
+  token,
+  days,
+}: {
+  token: string
+  days?: number
+}) {
+  const allTransactions = []
+
+  let daysAgo
+  if (days) {
+    daysAgo = Math.floor(Date.now() / 1000) - days * 24 * 60 * 60
+  }
+
+  let response = await fetchFirstTransactions(token)
+  let transactions = response?.edges || []
+  let pageInfo = response?.pageInfo
+
+  allTransactions.push(...transactions)
+
+  let lastTransactionDate =
+    transactions.length > 0 ? transactions[transactions.length - 1].node.createdAt : 0
+
+  while (pageInfo?.hasNextPage && (!days || (daysAgo && lastTransactionDate > daysAgo))) {
+    const nextCursor = pageInfo.endCursor
+    response = await fetchPaginatedTransactions(token, "next", nextCursor)
```

Comment:
we are using object instead of positional argument in our codebase

Reply:
the only except is if there is a single argument

Reply:
Yes, I am aware. This is an old function that we are using in other places as well. I will make a PR to convert all these to take an object.

Reply:
we don't necessarily need to rush into adding new features like the balance history graph

I think it's good to focus on code quality for the time being, which means we can start with this refactoring and rebase this PR when the refactoring PR is merged to main.